coroutines-extra/src/main/scala/AsyncAwait.scala

```diff
@@ -0,0 +1,101 @@
+package org.coroutines.extra
+
+
+
+import org.coroutines._
+import scala.annotation.unchecked.uncheckedVariance
+import scala.concurrent._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.{ Success, Failure }
+import scala.language.experimental.macros
+import scala.reflect.macros.whitebox.Context
+
+
+
+object AsyncAwait {
+  /** A wrapper class for a variable of type `T`.
+   *
+   *  Used inside `await` to wrap the results of asynchronous computations.
+   */
+  class Cell[+T] {
```

Comment:
Despite the example that we have in our documentation, it's not entirely clear to me if `Cell` is necessary.
What if `await` yielded a `Future` object instead of a `Cell`? It seems like the `Future` has all the information that you are copying into `Cell`'s field `x` (see method `value` on `Future`).